test/bmc/redfish_test_helper.rb

```diff
@@ -0,0 +1,101 @@
+
+  SYSTEMS_DATA = {
+    "@odata.id" => "/redfish/v1/Systems",
+    "Members" => [
+      { "@odata.id" => "/redfish/v1/Systems/TESTSYSTEM" },
+    ],
+  }.freeze
+
+  SYSTEM_DATA = {
+    "@odata.id" => "/redfish/v1/Systems/TESTSYSTEM",
+    "Manufacture" => "Server Vendor",
+    "IndicatorLED" => "Off",
+    "PowerState" => "On",
+    "Boot" => {
+      "BootSourceOverrideTarget" => "Pxe",
+    },
+    "Model" => "TestServer A",
+    "SerialNumber" => "AA645",
+    "AssetTag" => "ax",
+    "Actions" => {
+      "#ComputerSystem.Reset" => {
+        "target" => "/redfish/v1/Systems/TESTSYSTEM/Actions/ComputerSystem.Reset",
+      },
+    },
+  }.freeze
+
+  MANAGERS_DATA = {
+    "@odata.id" => "/redfish/v1/Managers",
+    "Members" => [
+      { "@odata.id" => "/redfish/v1/Managers/TESTBMC" },
+    ],
+  }.freeze
+
+  MANAGER_DATA = {
+    "@odata.id" => "/redfish/v1/Managers/TESTBMC",
+    "FirmwareVersion" => "v1.0.0",
+    "UUID" => "dff64837-f76b-40c2-b1a2-4717a78d828d",
+    "EthernetInterfaces" => {
+      "@odata.id" => "/redfish/v1/Managers/TESTBMC/EthernetInterfaces",
+    },
+  }.freeze
+
+  ETHERNET_INTERFACES_DATA = {
+    "@odata.id" => "/redfish/v1/Managers/TESTBMC/EthernetInterfaces",
+    "Members" => [
+      { "@odata.id" => "/redfish/v1/Managers/TESTBMC/EthernetInterfaces/BMC" },
+    ],
+  }.freeze
+
+  BMC_NIC_DATA = {
+    "@odata.id" => "/redfish/v1/Managers/TESTBMC/EthernetInterfaces/BMC",
+    "MACAddress" => "52:54:00:01:23:45",
+    "IPv4Addresses" => [
+      { "Address" => "192.168.124.123",
+        "Gateway" => "192.168.124.1",
+        "SubnetMask" => "255.255.255.0",
+        "AddressOrigin" => "DHCP" },
+    ],
+    "VLAN" => {
+      "VLANEnable" => true,
+      "VLANId" => 10,
+    },
+  }.freeze
+
+  def mask_redfish_acceess(protocol: "https", host: "host")
+    # root
+    stub_request(:get, "#{protocol}://#{host}#{ROOT_DATA['@odata.id']}").
+      to_return(status: 200, body: JSON.generate(ROOT_DATA))
+    # root.systems
+    stub_request(:get, "#{protocol}://#{host}#{SYSTEMS_DATA['@odata.id']}").
+      to_return(status: 200, body: JSON.generate(SYSTEMS_DATA))
+    # root.system
+    stub_request(:get, "#{protocol}://#{host}#{SYSTEM_DATA['@odata.id']}").
+      to_return(status: 200, body: JSON.generate(SYSTEM_DATA))
+    # root.managers
+    stub_request(:get, "#{protocol}://#{host}#{MANAGERS_DATA['@odata.id']}").
+      to_return(status: 200, body: JSON.generate(MANAGERS_DATA))
+    # root.manager
+    stub_request(:get, "#{protocol}://#{host}#{MANAGER_DATA['@odata.id']}").
+      to_return(status: 200, body: JSON.generate(MANAGER_DATA))
+    # root.manager.ethernetinterfaces
+    stub_request(:get, "#{protocol}://#{host}#{ETHERNET_INTERFACES_DATA['@odata.id']}").
+      to_return(status: 200, body: JSON.generate(ETHERNET_INTERFACES_DATA))
+    # bmcnic
+    stub_request(:get, "#{protocol}://#{host}#{BMC_NIC_DATA['@odata.id']}").
+      to_return(status: 200, body: JSON.generate(BMC_NIC_DATA))
+  end
```

Comment:
```suggestion
    def mask_redfish_acceess(protocol: "https", host: "host")
    # root
    stub_request(:get, "#{protocol}://#{host}#{ROOT_DATA['@odata.id']}").
      to_return(status: 200, body: JSON.generate(ROOT_DATA))
    # root.systems
    stub_request(:get, "#{protocol}://#{host}#{SYSTEMS_DATA['@odata.id']}").
      to_return(status: 200, body: JSON.generate(SYSTEMS_DATA))
    # root.system
    stub_request(:get, "#{protocol}://#{host}#{SYSTEM_DATA['@odata.id']}").
      to_return(status: 200, body: JSON.generate(SYSTEM_DATA))
  end
```

Looks like not all of these are required

Reply:
Yes, there is non-called mock. However the reason to write unused mock in this PR is to create common Redfish test code.
This same unit test preparation code is to be used at https://github.com/theforeman/smart-proxy/pull/916 & https://github.com/theforeman/smart-proxy/pull/917 and (with hope) future Redfish related PRs.

So unused mock is included here.

Reply:
@vanou I would argue with the same reasoning as in the [capability comment](https://github.com/theforeman/smart-proxy/pull/919#discussion_r2340232396). 

PR should introduce code that is only relevant to the implemented feature. It's easier for cherry-picking, debugging, and investigating the git log.

If there are constants needed for other PRs, they should be added from those PRs.

Reply:
@stejskalleos 
Thanks for reply. I understand what you concern. I agree it's easier for cherry-picking, debugging, investigating, etc. if we remove non-relevant code.